### DISCUSSION TITLE
TFT_eSPI refuses to work with M5Dial no matter what.

### DISCUSSION THREAD

**Discussion by vishaljagdish** [Bodmer/TFT_eSPI#3548] [Q&A - General] [answered]
Hello, I am trying to run an LVGL example using TFT_eSPI on my M5Dial (by M5Stack) using Arduino IDE. For some reason, it does not display anything no matter what I do. For context, the M5Dial uses a GC9A01A Driver to drive its display. It works perfectly when I run LVGL using the Adafruit GC9A01A library. My TFT eSPI config files are as follows:

User_setup_select.h
`
#include <User_Setup.h> 
#include <User_Setups/Setup200_GC9A01.h>  
//#include <User_Setups/Setup46_GC9A01_ESP32.h> 
`

i also tried Setups46 and its corresponding config file to no avail. 

User_Setup.h
`
#define USER_SETUP_INFO "User_Setup"
#define GC9A01_DRIVER
 #define TFT_WIDTH  240 // ST7789 240 x 240 and 240 x 320
 #define TFT_HEIGHT 240 // GC9A01 240 x 240
  `
  
 Setup200_GC9A01.h 
`
#define USER_SETUP_ID 200

#define GC9A01_DRIVER

// For ESP32 Dev board (only tested with GC9A01 display)
// The hardware SPI can be mapped to any pins

#define TFT_MOSI 5 // In some display driver board, it might be written as "SDA" and so on.
#define TFT_SCLK 6
#define TFT_CS   7  // Chip select control pin
#define TFT_DC   4  // Data Command control pin
#define TFT_RST  8  // Reset pin (could connect to Arduino RESET pin)
#define TFT_BL   9  // LED back-light
`
[Here](https://m5stack.oss-cn-shenzhen.aliyuncs.com/resource/docs/products/core/M5Dial/Sch_M5Dial.pdf) is the schematic for your reference

As a pre-test, I tested TFT_eSPI with a very basic example. I found that when I used `#include "M5Dial.h`, which is required to initialize the board, i got these errors:

- esp32/common.hpp:167:32: error: 'SeekMode' has not been declared
  167 |     bool seek(uint32_t offset, SeekMode mode) { return _fp->seek(offset, mode); }
- 'SeekCur' was not declared in this scope; did you mean 'fs::SeekCur'?
- packages/esp32/hardware/esp32/3.0.5/libraries/FS/src/FS.h:42:3: note: 'fs::SeekCur' declared here
- invalid conversion from 'int' to 'fs::SeekMode' [-fpermissive]
- In member function 'bool lgfx::v1::DataWrapperT<fs::File>::seek(uint32_t, int)':

The test code was 
`
#include <SPI.h>
#include <TFT_eSPI.h>
#include "M5Dial.h"

// Use hardware SPI
TFT_eSPI tft = TFT_eSPI();

void setup(){
  auto cfg = M5.config();
  M5Dial.begin(cfg, true, false);
  tft.init();
  tft.fillScreen(TFT_GREEN);
}

void loop(){
  
}
`
and similar errors related to the filesystem. Navigating to common.hpp (a part of M5GFX source code - used by M5Dial) and adding fs:: before SeekCur, SeekSet and SeekMode resolved the compilation error and successfully ran the code. However, the display was still off. Its a bit frustrating because I cannot seem to get it to work for seemingly no reason, while at the same time getting it to work perfectly with the Adafruit library. Any help/suggestions would be highly appreciated.

**Comment by Basitadam** [comment]
Hi @vishaljagdish,
MCU is ESP32-S3 right?
What do you see in Arduino Serial Monitor? is it in a boot loop?

**Reply by vishaljagdish** [reply]
Hello @Basitadam, 

Thank you for replying. Yes, sometimes its stuck in an infinite loop (the backlight seems to be turning on and off as well) and sometimes the display just does not show anything.

**Reply by Basitadam** [reply, marked as answer]
Can you add this line to your setup (Setup200_GC9A01.h) file:

``` 
#define USE_HSPI_PORT
```

**Reply by Basitadam** [reply]
By the way, you need to have only one setup file uncommented. In "User_setup_select.h" file, comment out "#include <User_Setup.h>" line and leave only "#include <User_Setups/Setup200_GC9A01.h>" line uncommented

**Reply by vishaljagdish** [reply]
This line worked!! Perfect! Thank you so much

What does this line do by the way?

I tried running the Animated_Eyes_2 Example. The refresh/rewrite rate seems to be a bit low. I don't know if that is normal

**Reply by Basitadam** [reply]
Line tells the library which SPI port of MCU to use. Not needed before but something changed in ESP32-IDF library, now needed for ESP32-S3 MCUs.
For faster refresh rate you can increase ```#define SPI_FREQUENCY  27000000```  up to 80000000 (for ESP32)

**Reply by vishaljagdish** [reply]
Understood. Thank you very much.

**Reply by Basitadam** [reply]
```HappyFace();```
